submissions/bmukha/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,73 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/bmukha/a-tiny-JS-world
+   Web app: https://bmukha.github.io/a-tiny-JS-world/
+   */
+
+class Character {
+  constructor(specie, name, gender, saying, friends, legs) {
+    this.specie = specie;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.saying = saying;
+    this.friends = friends;
+  }
+
+  props() {
+    return ['specie', 'name', 'gender', 'legs', 'hands', 'saying', 'friends']
```

Comment:
I kinda thought that preserving the order of props in the resulting string across all entities is a key requirement of this task. Am I wrong? Because I just can't see the easy way to achieve this goal using super. Should I handle inherited props using super, concatenate own properties to the end of the string and call it a day or I must come up with some kind of solution to put them in the correct places in the string? Another approach I can think of - is to make separate props array for each class. This way I can preserve desired order of props for every class, and I don't need to use the super keyword at all, I can handle everything by calling a slightly modified inherited method from the very first class. Which approach should I take?

Reply:
> Should I handle inherited props using super, concatenate own properties to the end of the string and call it a day or I must come up with some kind of solution to put them in the correct places in the string?

Let's put aside the solution where we may want **insert** child's props between parents' props. Let's only append or prepend child's props. For the sake of simplicity